src/Type/Php/ReflectionExtensionConstructorThrowTypeExtension.php

```diff
@@ -0,0 +1,45 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Type\Php;
+
+use PhpParser\Node\Expr\StaticCall;
+use PHPStan\Analyser\Scope;
+use PHPStan\Reflection\MethodReflection;
+use PHPStan\Type\DynamicStaticMethodThrowTypeExtension;
+use PHPStan\Type\NeverType;
+use PHPStan\Type\Type;
+use PHPStan\Type\TypeCombinator;
+use PHPStan\Type\TypeUtils;
+use ReflectionExtension;
+
+class ReflectionExtensionConstructorThrowTypeExtension implements DynamicStaticMethodThrowTypeExtension
+{
+
+	public function isStaticMethodSupported(MethodReflection $methodReflection): bool
+	{
+		return $methodReflection->getName() === '__construct' && $methodReflection->getDeclaringClass()->getName() === ReflectionExtension::class;
+	}
+
+	public function getThrowTypeFromStaticMethodCall(MethodReflection $methodReflection, StaticCall $methodCall, Scope $scope): ?Type
+	{
+		if (count($methodCall->args) < 1) {
+			return $methodReflection->getThrowType();
+		}
+
+		$valueType = $scope->getType($methodCall->args[0]->value);
+		foreach (TypeUtils::getConstantStrings($valueType) as $constantString) {
+			if (!extension_loaded($constantString->getValue())) {
```

Comment:
`extension_loaded` isn't a useful concept because PHPStan does not require extensions to be loaded during analysis. It's best to skip this extension altogether.